sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -117,6 +121,25 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 messageArgs: MemberNameToMessageArguments[memberAccessNameName]));
         }
 
+        private static bool IsResultInContinueWithCall(string memberAccessName, MemberAccessExpressionSyntax memberAccess) =>
+            memberAccessName == ResultName &&
+            memberAccess.Expression is IdentifierNameSyntax identifierNameSyntax &&
+            identifierNameSyntax.GetName() is {} identifierName &&
+            memberAccess.FirstAncestorOrSelf<InvocationExpressionSyntax>(invocation => IsContinueWithCallWithArgumentName(invocation, identifierName)) != null;
+
+        private static bool IsContinueWithCallWithArgumentName(InvocationExpressionSyntax invocation, string argumentName) =>
+            invocation.Expression.NameIs(ContinueWithName) &&
+            invocation.ArgumentList.Arguments.Any(argument => IsLambdaExpressionWithArgumentName(argument.Expression, argumentName));
+
+        private static bool IsLambdaExpressionWithArgumentName(ExpressionSyntax expression, string argumentName) =>
+            expression switch
+            {
+                SimpleLambdaExpressionSyntax lambda => lambda.Parameter.Identifier.ValueText == argumentName,
+                ParenthesizedLambdaExpressionSyntax parenthesizedLambda =>
```

Comment:
I was wondering if the `Any` is really needed and it is. There are so many overloads of `ContinueWith`, I suggest to add some of them to tests. We already have overload for `Func<Task<T>, TResult>`.
Please add tests
```
// Action<Task<T>>
var a= task.ContinueWith(completedTask =>
{
    var result = completedTask.Result; // Compliant, task is already completed at this point.
});
// Action<Task<T>, object>
var b = task.ContinueWith((completedTask, state) =>
{
    var result = completedTask.Result; // Compliant, task is already completed at this point.
}, null);
// Func<Task<T>, object, TResult>
var c = task.ContinueWith((completedTask, state) =>
{
    return completedTask.Result; // Compliant, task is already completed at this point.
}, null);
```